training/metrics.py

```diff
@@ -27,59 +25,91 @@ def count_trainable_params(model):
   total_trainable_params = int(sum(np.prod(p.shape.as_list()) for p in object_identity.ObjectIdentitySet(weights)))
   return total_trainable_params
 
-
-def _count_flops_conv2d(layer):
-  if layer.data_format == "channels_first":
-    input_channels = layer.input_shape[1]
-    output_channels, h, w, = layer.output_shape[1:]
-  elif layer.data_format == "channels_last":
-    input_channels = layer.input_shape[3]
-    h, w, output_channels = layer.output_shape[1:]
-  w_h, w_w = layer.kernel_size
-
-  n_mult = h * w * output_channels * input_channels * w_h * w_w
-  n_add = w_h * w_w * input_channels * h * w * output_channels
-
-  flops = n_mult + n_add
+def _linear_layer(layer, N):
+  """
+  Note: This calculates the FLOPs for the unoptimized implementation of any Algebra
+  """
+  input_shape = layer.input_shape
+  output_shape = layer.output_shape
+  if len(input_shape) == 4: # 2D Conv and DepthWise
+    if layer.data_format == "channels_first":
+      input_channels = input_shape[1]
+      output_channels, h, w, = output_shape[1:]
+    elif layer.data_format == "channels_last":
+      input_channels = input_shape[3]
+      h, w, output_channels = output_shape[1:]
+    w_h, w_w = layer.kernel_size
+    if layer.name.lower() == "depthwise_conv2d": 
```

Comment:
this condition is unreliable : users can change the name of the layer without control